### ISSUE TITLE
Parallax Occlusion has gone haywire once more

### ISSUE THREAD

**Issue by DoctorAnsem** [SleepKiller/shaderpatch#91]
[This](https://drive.google.com/open?id=1_HJN5nXAHZ0fo6lsjAeQk6Tc8QJvGC3_) should explain it all. There are a lot of instances where near edges, the parallax occlusion turns the surface into liquid. Terrains as well are very vulnerable to this.

**Comment by SleepKiller** [comment]
Yeah I noticed it on terrains as well. I'm going to try taking one more look at it before I release the next version to see if I can spot anything I've done wrong. To be honest I don't know if I'll be able to solve it or not but I'll certainly try.

In the meantime I think Gogie found a way to work around this issue (or a similar one) for regular models by editing the normals of a model, here's a screenshot of his message. (Apologies to anyone reading this on a high DPI display.)
![image](https://user-images.githubusercontent.com/5582558/65815770-cb20f180-e247-11e9-9002-4472632966ef.png)

Of course I'm not a modeler so I only understand about half of what he's talking about but it looks like it solves it.

**Comment by DoctorAnsem** [comment]
Is that actually possible for BF2? I thought ZETool automatically recalculates normals during mesh export, based on UV islands and/or polygon angles.

**Comment by SleepKiller** [comment]
I believe it must be, the top and bottom screenshots are from before and after he applied his fix. I could seek clarification that he was using XSIZETools to export it if you'd like?

**Comment by DoctorAnsem** [comment]
I guess I'll try it. Still, that's a _lot_ of work if every problematic model needs to be redone like that. It might not even be worth the effort, but we'll see.

**Comment by SleepKiller** [comment]
I'm afraid I've been unable to figure out what's wrong for now. It looks like Parallax Occlusion Mapping will have to be consigned to being only for relatively flat surfaces for the time being.
